node/pkg/reporter/main_test.go

```diff
@@ -2,172 +2,69 @@
-}
-
-func insertSampleData(ctx context.Context) (*TmpData, error) {
-	var tmpData = new(TmpData)
-
-	signHelper, err := helper.NewSigner(ctx)
-	if err != nil {
-		return nil, err
-	}
-	proofTime := time.Now()
-
-	tmpConfig, err := db.QueryRow[Config](ctx, InsertConfigQuery, map[string]any{"name": "test-aggregate", "submit_interval": TestInterval, "fetch_interval": TestInterval, "aggregate_interval": TestInterval})
-	if err != nil {
-		log.Error().Err(err).Msg("error inserting config 0")
-		return nil, err
-	}
-	tmpData.config = tmpConfig
-
-	err = db.QueryWithoutResult(ctx, InsertConfigQuery, map[string]any{"name": "test-aggregate-2", "submit_interval": TestInterval * 2, "fetch_interval": TestInterval, "aggregate_interval": TestInterval})
-	if err != nil {
-		log.Error().Err(err).Msg("error inserting config 1")
-		return nil, err
+func getConfigUrl() string {
+	configUrl := os.Getenv("ADMIN_ENDPOINT")
+	if configUrl == "" {
+		configUrl = "http://100.65.92.37:3030/api/v1"
 	}
 
-	key := keys.GlobalAggregateKey(tmpConfig.ID)
-	data, err := json.Marshal(map[string]any{"configId": tmpConfig.ID, "value": int64(15), "round": int64(1), "timestamp": proofTime})
-	if err != nil {
-		return nil, err
-	}
-	db.Set(ctx, key, string(data), time.Duration(10*time.Second))
-
-	tmpGlobalAggregate, err := db.QueryRow[GlobalAggregate](ctx, InsertGlobalAggregateQuery, map[string]any{"config_id": tmpConfig.ID, "value": int64(15), "round": int64(1), "timestamp": proofTime})
-	if err != nil {
-		return nil, err
-	}
-	tmpData.globalAggregate = tmpGlobalAggregate
-
-	rawProof, err := signHelper.MakeGlobalAggregateProof(int64(15), proofTime, "test-aggregate")
-	if err != nil {
-		return nil, err
-	}
-	tmpData.proofBytes = bytes.Join([][]byte{rawProof}, nil)
-	tmpData.proofTime = proofTime
-
-	err = db.QueryWithoutResult(ctx, "INSERT INTO proofs (config_id, round, proof) VALUES (@config_id, @round, @proof)", map[string]any{"config_id": tmpConfig.ID, "round": int64(1), "proof": bytes.Join([][]byte{rawProof}, nil)})
-	if err != nil {
-		return nil, err
-	}
-
-	rdbProof := Proof{
-		ConfigID: tmpConfig.ID,
-		Round:    int32(1),
-		Proof:    bytes.Join([][]byte{rawProof}, nil),
-	}
-	rdbProofData, err := json.Marshal(rdbProof)
-	if err != nil {
-		return nil, err
-	}
+	return configUrl + "/config"
+}
 
-	err = db.Set(ctx, keys.ProofKey(tmpConfig.ID, 1), string(rdbProofData), time.Duration(10*time.Second))
+func fetchConfigs(ctx context.Context) ([]Config, error) {
+	configUrl := getConfigUrl()
+	configs, err := request.Request[[]Config](request.WithEndpoint(configUrl), request.WithTimeout(10))
 	if err != nil {
 		return nil, err
 	}
-
-	return tmpData, nil
+	return configs, nil
 }
 
-func setup(ctx context.Context) (func() error, *TestItems, error) {
-	var testItems = new(TestItems)
-
-	mb := bus.New(10)
-	testItems.messageBus = mb
-
-	admin, err := utils.Setup(utils.SetupInfo{
-		Version: "",
-		Bus:     mb,
-	})
+func setup(ctx context.Context) error {
```

Comment:
I'd like to recommend you to set things up in each test and clean up after each test rather than setting up single time to be used in whole test. it's good to have each tests in clean slate rather than reusing the settings.

And if the data is not cleaned up after test, the configs left in db might lead to failures in other package tests

Reply:
I agree with your suggested design in general, but the reason I decided to go with. the current design is because the configs are added once before tests start and then removed after all tests are passed. Since none of the tests mutate the config data (they only read), I thought it might be redundant to add and delete for each `reporter` test. But I'm making sure the added configs are deleted right after all `reporter` tests pass.

Reply:
good point, I'll update as you've suggested 🙏

Reply:
done!

Reply:
please push the branch before you make it resolved 🙏